### PR TITLE
[#52921] Configure HTTPX timeouts.

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/52921

- Configure HTTPX timeouts
- Reuse HTTPX session though lifetime of a process.
- Enable HTTPX persistent connections(with reusage of HTTPX session it should work as expected)
- Add HTTPXApsignal plugin to HTTPX if Appsignal enabled to track external requests. [Sample](https://appsignal.com/openproject-gmbh/sites/65d5f46a83eb679274de83f9/performance/incidents/30/samples/65d5f46a83eb679274de83f9-176837881530755131351708712520).
- Run Storages::ManageNextcloudIntegrationCronJob once per hour.

### CODE REVIEW

File: lib/open_project.rb [hunk 1]
```diff
@@ -46,8 +47,23 @@ def self.logger
   end
 
   def self.httpx
-    HTTPX
-      .plugin(:basic_auth)
-      .plugin(:webdav)
+    # It is essential to reuse HTTPX session if persistent connections are enabled.
+    # Otherwise for every request there will be a new connections.
+    # And old connections will not be closed properly which could lead to EMFILE error.
+    Thread.current[:httpx_session] ||= begin
```
Comment (mereghost) [inline lib/open_project.rb:53]: Won't we run into issues as we did with the workarounds around DelayedJob?

Comment (ba1ash) [reply lib/open_project.rb:53]: You mean in saas environment?
In this case it should be all right to share a session between multiple tenants...

Comment (mereghost) [reply lib/open_project.rb:53]: True that! :D

File: modules/storages/app/workers/storages/manage_nextcloud_integration_cron_job.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ class ManageNextcloudIntegrationCronJob < Cron::CronJob
 
     queue_with_priority :low
 
-    self.cron_expression = '*/5 * * * *'
+    self.cron_expression = '1 * * * *'
```
Comment (mereghost) [inline modules/storages/app/workers/storages/manage_nextcloud_integration_cron_job.rb:37]: First minute of the hour :) Classy. 🧐

**Review by mereghost — APPROVED** [review]
LGTM.

Can't wait to see the metrics kicking in.

**Pull request merged**